src/app/WriteHandler.cpp

```diff
@@ -202,6 +206,46 @@ CHIP_ERROR WriteHandler::SendWriteResponse(System::PacketBufferTLVWriter && aMes
     return err;
 }
 
+void WriteHandler::DeliverListWriteBegin(const ConcreteAttributePath & aPath)
+{
+    if (auto * attrOverride = GetAttributeAccessOverride(aPath.mEndpointId, aPath.mClusterId))
+    {
+        attrOverride->OnListWriteBegin(aPath);
+    }
+}
+
+void WriteHandler::DeliverListWriteEnd(const ConcreteAttributePath & aPath, bool writeWasSuccessful)
+{
+    if (auto * attrOverride = GetAttributeAccessOverride(aPath.mEndpointId, aPath.mClusterId))
+    {
+        attrOverride->OnListWriteEnd(aPath, writeWasSuccessful);
+    }
+}
+
+void WriteHandler::DeliverFinalListWriteEnd(bool writeWasSuccessful)
+{
+    if (mProcessingAttributePath.HasValue() && mProcessingAttributeIsList)
+    {
+        DeliverListWriteEnd(mProcessingAttributePath.Value(), writeWasSuccessful);
+    }
+    mProcessingAttributePath.ClearValue();
+}
+namespace {
+
+bool ShouldReportListWriteEnd(const Optional<ConcreteAttributePath> & previousProcessed, bool previousProcessedAttributeIsList,
+                              const ConcreteAttributePath nextAttribute)
+{
+    return previousProcessed.HasValue() && previousProcessedAttributeIsList && previousProcessed.Value() != nextAttribute;
```

Comment:
OK, so in my example above there will be one single list write end, after the second list write?